### DISCUSSION TITLE
Migrate adminerevo to Bootstrap ?

### DISCUSSION THREAD

**Discussion by LionelLaffineur** [adminerevo/adminerevo#5] [Polls]
Shall we migrate adminerevo to Bootstrap ?

Pros / cons ?  Please comment your answer.

**Comment by nitodico** [comment]
How will affect the compatiblity with the CSS themes made by the community?

**Reply by LionelLaffineur** [reply]
This could make them unusable.  It is indeed something to take into account.

**Reply by nitodico** [reply]
Maybe add it to issues with an enhacement tag and look in the future

**Reply by Roy-Orbison** [reply]
It's probably not a bad idea to break compatibility with existing themes if it entails a better UX and easier theme creation/updates. Most are outdated anyway.

**Comment by Roy-Orbison** [comment]
Bootstrap's not bad, but isn't it a relatively heavy addition to a tool that needs to be snappy? Not all Internet connections are equal. It also bloats the markup with extra classes, and wrapping elements. I think in many cases Adminer's markup could actually be reduced, and have more concise CSS applied. [A CSS reset](https://www.joshwcomeau.com/css/custom-css-reset/) would aid consistency.

I agree that Adminer would benefit from a redesign as, although compatibility with older technologies on the server side is a goal, its client-side environment is much less likely to be lagging behind. Some judicious use of flexbox, and making it properly responsive could really modernise it, and even improve the ability to skin it. It's not really responsive currently because smaller viewports require the user scroll horizontally and vertically because Jakub wrote styles based on `device-width`, unlike almost every other site on the planet.

It would also be good to have a plan re skins because the versions in Adminer's main repo are mostly either stale or behind their authors' versions, which leaves a bad UX and discourages contributions.

FYI, Bootstrap's current (v5.3) sizes:

|File|Compressed (Brotli)|Uncompressed|
|---|---|---|
|[bootstrap.min.css](https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css)|~35kb|~233kb|
|[bootstrap.bundle.min.js](https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/js/bootstrap.bundle.min.js)|~25kb|~81kb|

There are lighter frameworks, like [Pure.css](https://purecss.io/).

**Reply by LionelLaffineur** [reply]
Hi @Roy-Orbison,
After internal discussion, we all agree that using Bootstrap in adminer wouldn't correctly serve the purpose of the product.
We are however a little in the dark about your idea of having adminer responsive whatever the framework or browser native.
Would you drop here some mockups on how you see the larger pages (like resultsets with many columns) to be responsive ?
It's **an idea** to start the first release of adminerevo with a new major release, where we would start over with a new layout, up-to-date with latest design standards, even if it breaks compatibility with existing themes.

**Reply by zzjin** [reply]
redesign +1, bootstrap? will consider too heavy too.

**Reply by Roy-Orbison** [reply]
It wasn't so much about making tabular data fully responsive, however [that is possible](https://stackoverflow.com/a/64706497/315024), it was about making the default design respond to viewport width instead of device width, and some neater markup.

Adminer's current element order is reasonably semantic, data first, then menu. I think keep that. However, it uses floats and margin tricks that were once necessary, but are now obsolete. There are a couple of skins that do a decent job of responsive design, so I included them in the WordPress plugin I put together that packages up Adminer. They are an adequate demo of what can be achieved.

**Reply by Roy-Orbison** [reply]
Re responsive tables: having those attributes on every data cell does increase the size of the markup, *but* it compresses very well on-the-fly because they're identical each row. Maybe responsive tables could be a pref held in a cookie like language is now? Not sure.

**Comment by mckaygerhard** [comment]
no cos means will be no more "lightweight" ..

**Reply by LionelLaffineur** [reply]
Yes, indeed. After readings, internal discussions, the idea has been closed.
I will close the discussion.
